src/components/VirtualFacilityActionsPopover.vue

```diff
@@ -0,0 +1,163 @@
+    </ion-list>
+  </ion-content>
+</template>
+  
+<script lang="ts">
+import {
+  alertController,
+  IonContent,
+  IonItem,
+  IonList,
+  IonListHeader,
+  popoverController
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { lockClosedOutline, lockOpenOutline } from 'ionicons/icons'
+import { translate } from '@hotwax/dxp-components'
+import { showToast } from '@/utils';
+import { FacilityService } from "@/services/FacilityService";
+import { hasError } from "@/adapter";
+import { mapGetters, useStore } from "vuex";
+import logger from "@/logger";
+
+export default defineComponent({
+  name: "VirtualFacilityActionsPopover",
+  components: {
+    IonContent,
+    IonItem,
+    IonList,
+    IonListHeader
+  },
+  computed: {
+    ...mapGetters({
+      virtualFacilities: 'facility/getVirtualFacilities',
+    })
+  },
+  props: ['facility'],
+  methods: {
+    async renameVirtualFacility() {
+      const alert = await alertController.create({
+        header: translate('Rename parking'),
+        inputs: [{
+          name: "facilityName",
+          value: this.facility.facilityName
+        }],
+        buttons: [{
+          text: translate('Cancel'),
+          role: "cancel"
+        },
+        {
+          text: translate('Apply'),
+          handler: (data) => {
+            const { facilityName } = data
+            popoverController.dismiss(facilityName)
+          }
+        }]
+      })
+      await alert.present()
+    },
+    async archiveVirtualFacility() {
+      try {
+        let facilityGroupId = await this.fetchArchiveGroup()
+
+        if (!facilityGroupId) {
+          facilityGroupId = await this.createArchiveGroup()
+        }
+        
+        if (!facilityGroupId) {
+          throw { message: translate('Failed to archive parking.') }
+        }
+
+        const resp = await FacilityService.addFacilityToGroup({
+          facilityId: this.facility.facilityId,
+          facilityGroupId
+        })
+
+        if (!hasError(resp)) {
+          const updatedVirtualFacilities = JSON.parse(JSON.stringify(this.virtualFacilities))
+            .filter((facility: any) => facility.facilityId !== this.facility.facilityId)
+          this.store.dispatch('facility/updateVirtualFacilities', updatedVirtualFacilities)
+          await this.store.dispatch('facility/fetchArchivedFacilities')
+          showToast(translate("Parking archived successfully."))
+        } else {
+          throw resp.data
+        }
+      } catch (error: any) {
+        showToast(error.message)
+        logger.error(error.message)
+      }
+
+      popoverController.dismiss()
+    },
+    async fetchArchiveGroup() {
+      let fetchedFacilityGroupId = ''
+      try {
+        const resp = await FacilityService.fetchFacilityGroup({
+          inputFields: {
+            facilityGroupId: 'ARCHIVE',
+          },
+          entityName: 'FacilityGroup',
+          viewSize: 1
+        })
+
+        if (hasError(resp)) {
```

Comment:
As we are going to create an Archive facility group if not exist, you can simply return the facilityGroupId if found.